envoy: Rebase to get gRPC timeout support.

jrajahalme opened the pull request:
Rebase to the upstream Envoy commit that adds support for
'max_grpc_timeout'. Pass the zero value for it, allowing infinite gRPC
timeouts. The actual timeout is derived from the 'grpc-timeout'
header, but if missing, it defaults to infinity.

TODO:
- [x] Make sure Istio GSG still works

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/4508)
<!-- Reviewable:end -->

rlenglet commented on pkg/envoy/server.go:
```diff
@@ -182,7 +182,8 @@ func StartXDSServer(stateDir string) *XDSServer {
 										"route": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{
-											"cluster": {Kind: &structpb.Value_StringValue{StringValue: "cluster1"}},
+											"cluster":          {Kind: &structpb.Value_StringValue{StringValue: "cluster1"}},
+											"max_grpc_timeout": {Kind: &structpb.Value_StructValue{StructValue: &structpb.Struct{Fields: map[string]*structpb.Value{}}}},
```
I'm curious what would happen if you set this option but the sidecar proxies are still running older versions of Envoy, which don't support this option?
I.e. are we backward compatible?
Could you please give it a shot, e.g. temporarily revert the Envoy SHA update and see whether Envoy barfs on that field?

rlenglet replied:
Nevermind. This is irrelevant for sidecar proxies, since their configuration will come from pilot-agent and/or pilot.
LGTM!

jrajahalme committed:
vendor: Update golang/protobuf

Ensure that code generated by latest protoc-gen-go compiles by
updating golang/protobuf dependency.

This may not be strictly necessary, but should not hurt either.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
envoy: Remove go_package options.

protoc-gen-go started complaining about inconsestent package names.
Fix this by removing 'option go_package' statements from cilium filter
protobuf files.

This was possibly caused by our dependency to "latest" protoc-gen-go.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
envoy: Update API location.

Upstream Envoy commit 1fdb6386c4bb ("Reconcile
envoyproxy/data-plane-api and envoyproxy/envoy (#3036)") moved Envoy
API protobufs and docs to the main Envoy repo. Bazel build system
keeps using the "envoy_api" name, but it is now implicitly defined to
whatever files exist in the main repo rather than to a specific commit
in the data-plane-api repo.

Change the path in Makefile.api to point to the api directory in the Envoy repo.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
envoy: Rebase to get gRPC timeout support.

Rebase to the upstream Envoy commit that adds support for
'max_grpc_timeout'. Pass the zero value for it, allowing infinite gRPC
timeouts. The actual timeout is derived from the 'grpc-timeout'
header, but if missing, it defaults to infinity.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

jrajahalme committed:
envoy: Update generated go protobufs

Update generated envoy API protobufs.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
